src/index.js

```diff
@@ -33,42 +37,98 @@ function fetchData(searchTerm) {
         .catch(function (error) { console.log(`${error.name}:\n${error.message}`); });
 }
 
-function decodeData(foundData) {
-    for (let foundArticle of foundData.query.search) {
-        // console.log(foundArticle.snippet)
 
+async function decodeData(foundData) {
+
+    prepareResultsSection()     // create section in html where resutls will be displayed
+
+    for (let foundArticle of foundData.query.search) {
+        // console.log(foundData)
         let articleTitle = foundArticle.title
+
         let articleText = foundArticle.snippet
-        .replace(/(<([^>]+)>)/gi, "")               // strip html tags from 
-        .slice(0, 130) + "...";                      // truncate text
-        
-        let articleUrl = 'https://en.wikipedia.org/?curid=' + Number(foundArticle.pageid)        //
+            .replace(/(<([^>]+)>)/gi, "")               // strip html tags from 
+            .slice(0, 130) + "...";                      // truncate text
+
+        let pageID = Number(foundArticle.pageid)
 
-        generateCard(articleTitle, articleText, articleUrl)
+        let articleUrl = 'https://en.wikipedia.org/?curid=' + pageID
+
+        let imgUrl = await getArticleImage(articleTitle, pageID)
+        // console.log(imgUrl)
+
+        generateCard(articleTitle, articleText, articleUrl, imgUrl)
     }
 }
 
+async function getArticleImage(title, pageID) {
+    let url = baseUrl + "?origin=*" + `&action=query&format=json&prop=pageimages&titles=${title}&pithumbsize=260`;        // "?origin=*" to resolve cors cross-origin request issues
+    let imgURL
 
-function generateCard(title, text, url) {
+    let res = await fetch(url)
+    let data = await res.json()
+    try {
+        imgURL = data.query.pages[String(pageID)].thumbnail.source
+        // console.log(data.query.pages[String(pageID)].thumbnail.source)
+    } catch (err) {
+        imgURL = '../img/default.jpg'
```

Comment:
I like that you've chosen to include a default image for any results without an image url, looks much better than just having alt text!